### PR TITLE
Feature: Add selected company to confirmation page

### PR DESCRIPTION
**Pull request by alaca**
## Description

[GIVE-493]

This PR adds company matching instructions on the Confirmation page.

## Affects

Confirmation page

## Visuals

Screen Capture_select-area_20240325082838

## Testing Instructions

Create a form with Double the Donation block
View form on the frontend
Select Company
Donate

The matching instructions should be visible at the bottom of the confirmation page

## Pre-review Checklist

-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-493]: https://stellarwp.atlassian.net/browse/GIVE-493?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/DoubleTheDonation/FormExtension/Actions/FieldScope/PaymentMeta.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ */
+class PaymentMeta
+{
+    /**
+     * @unreleased
+     */
+    public function __invoke(): Closure
+    {
+        // Scope callback
+        return function (DoubleTheDonationField $field, $company, Donation $donation) {
```
Comment (jonwaldstein) [inline src/DoubleTheDonation/FormExtension/Actions/FieldScope/PaymentMeta.php:22]: Woah this is an interesting approach. I was imaging the invoke method would receive the callback params - however I suppose if this is being passed directly to the scope than we would need the return callback.

File: src/DoubleTheDonation/FormExtension/ServiceProvider.php [hunk 2]
```diff
@@ -28,5 +29,6 @@ public function boot(): void
         Hooks::addAction('givewp_donation_form_enqueue_scripts', LoadAssets::class, 'donationForm');
 
         Hooks::addFilter('givewp_donation_form_block_render_givewp/dtd', ConvertBlockToField::class, '__invoke', 10, 4);
+        Hooks::addFilter('givewp_donation_confirmation_page_field_label_for_dtd', DisplayFieldLabel::class, '__invoke', 10, 3);
```
Comment (jonwaldstein) [inline src/DoubleTheDonation/FormExtension/ServiceProvider.php:32]: filter for the win 🦾

**Review by jonwaldstein — APPROVED** [review]
@alaca great job!

File: src/DoubleTheDonation/FormExtension/Actions/FieldScope/PaymentMeta.php (hunk 1 above)
Comment (alaca) [reply src/DoubleTheDonation/FormExtension/Actions/FieldScope/PaymentMeta.php:22]: I want to extend the Fields API to directly support passing an action class. Currently, you can only pass closures and strings. That's okay if you have really simple stuff to handle like updating one meta field, but anything more complex simply does not look good on the field definition.

**Pull request merged**